psi4/src/psi4/libmints/twobody.cc

```diff
@@ -166,38 +166,9 @@ double TwoBodyAOInt::shell_pair_max_density(int M, int N) const {
     return D_max;
 }
 
-// Haser 1989 Equations 6 to 14
-bool TwoBodyAOInt::shell_significant_density(int M, int N, int R, int S) {
-
-    // Maximum density matrix equation
-    double max_density = 0.0;
-
-    // Equation 6 (RHF Case)
-    if (max_dens_shell_pair_.size() == 1) {
-        max_density = std::max({4.0 * max_dens_shell_pair_[0][M * nshell_ + N], 4.0 * max_dens_shell_pair_[0][R * nshell_ + S], 
-            max_dens_shell_pair_[0][M * nshell_ + R], max_dens_shell_pair_[0][M * nshell_ + S],
-            max_dens_shell_pair_[0][N * nshell_ + R], max_dens_shell_pair_[0][N * nshell_ + S]});
-    } else { // UHF and ROHF
-        // J-like terms
-        double D_MN = max_dens_shell_pair_[0][M * nshell_ + N] + max_dens_shell_pair_[1][M * nshell_ + N];
-        double D_RS = max_dens_shell_pair_[0][R * nshell_ + S] + max_dens_shell_pair_[1][R * nshell_ + S];
-
-        // K-like terms
-        double D_MR = std::max(max_dens_shell_pair_[0][M * nshell_ + R], max_dens_shell_pair_[1][M * nshell_ + R]);
-        double D_MS = std::max(max_dens_shell_pair_[0][M * nshell_ + S], max_dens_shell_pair_[1][M * nshell_ + S]);
-        double D_NR = std::max(max_dens_shell_pair_[0][N * nshell_ + R], max_dens_shell_pair_[1][N * nshell_ + R]);
-        double D_NS = std::max(max_dens_shell_pair_[0][N * nshell_ + S], max_dens_shell_pair_[1][N * nshell_ + S]);
-
-        max_density = std::max({2.0 * D_MN, 2.0 * D_RS, D_MR, D_MS, D_NR, D_NS});
-    }
-
-    // Square of Cauchy-Schwarz Q_MN terms (Eq. 13)
-    double mn_mn = shell_pair_values_[N * nshell_ + M];
-    double rs_rs = shell_pair_values_[S * nshell_ + R];
-
-    // The density screened ERI bound (Eq. 6)
-    return (mn_mn * rs_rs * max_density * max_density >= screening_threshold_squared_);
-}
+double TwoBodyAOInt::shell_pair_max_density(int i, int M, int N) const  {
```

Comment:
Should we move `shell_pair_max_density` to JK in this PR or the next? I don't think this belongs in `TwoBodyAOInt`

Reply:
My plan was to move this whenever update_density gets moved, which I'm planning on being the next PR.